src/internal/Subscriber.ts

```diff
@@ -167,9 +167,9 @@ export class SafeSubscriber<T> extends Subscriber<T> {
     // Once we set the destination, the superclass `Subscriber` will
     // do it's magic in the `_next`, `_error`, and `_complete` methods.
     this.destination = {
```

Comment:
Ultimately, the issue is from https://github.com/ReactiveX/rxjs/commit/1b07686a1f79bb5b994ea85795fe3af868f07a4b,
When that was refactored, we removed the pieces that caught errors in user-provided handler functions, and then handled them appropriately.